### PR TITLE
Overwrite symlinked files & always save with --save flag

### PR DESCRIPTION
**Pull request by millzpaugh**
This PR: 
- saves a package to `mason-versions.ini` regardless of whether a package has already been installed #42 
- overwrites existing symlinked files

### CODE REVIEW

File: lib/symlink.js [hunk 1]
```diff
@@ -6,10 +6,12 @@ var filterFunc = function(src, dest) {
   if (src.indexOf('mason.ini') > -1) {
     return false;
   } else if (fs.lstatSync(src).isFile()) {
+    if (fs.existsSync(dest)){
+      fse.removeSync(dest);
```
Comment (springmeyer) [inline lib/symlink.js]: Why is this needed? Does `fs.symlinkSync(src, dest);` throw if the symlink already exists?

File: test/symlink.test.js [hunk 2]
```diff
@@ -97,6 +97,23 @@ test('[symlink] fails to create symlink - directory not found', function(assert)
   });
 });
 
+test('[symlink] overwrites existing files', function(assert) {
+  var src = path.join(__dirname + '/fixtures/', 'config.cpp');
+  var dst = path.join(__dirname + '/fixtures/', 'config-copy.cpp');
```
Comment (springmeyer) [inline test/symlink.test.js]: How about using much more simple files - like text files with a single word in them? This would make the test more minimal increasing signal to noise (since protozero code specifics not matter)

File: lib/symlink.js (hunk 1 above)
Comment (millzpaugh) [reply lib/symlink.js]: @springmeyer ya - `fs.symlinkSync` doesn't have an `overwrite` option, it just errors if a file already exists.

**Comment by millzpaugh** [comment]
@springmeyer this is a great catch - the file writer checks for the `[headers]` or `[compiled]` category depending on whether the library is a compiled or header-only. Then, it appends the package underneath the category. Since there aren't any `compiled` libraries being used in vtquery, `[compiled]` doesn't exist in vtquery's `mason-versions.ini`. I'll need to add code that handles the case where either of the respective categories don't exist.

**Comment by springmeyer** [comment]
Looks good, `--save` command now works for me locally. Nice work!
